tests/Unit/ParallelAlgorithms/LinearSolver/Schwarz/Test_OverlapHelpers.cpp

```diff
@@ -5,11 +5,136 @@
 
 #include "DataStructures/DataVector.hpp"
 #include "DataStructures/Index.hpp"
+#include "DataStructures/Tags/TempTensor.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Structure/Direction.hpp"
 #include "ErrorHandling/Error.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
 #include "ParallelAlgorithms/LinearSolver/Schwarz/OverlapHelpers.hpp"
+#include "Utilities/TMPL.hpp"
 
 namespace LinearSolver::Schwarz {
 
+template <size_t Dim>
+void test_data_on_overlap_consistency(const Index<Dim>& volume_extents,
+                                      const size_t overlap_extent,
+                                      const Direction<Dim>& direction) {
+  MAKE_GENERATOR(generator);
+  std::uniform_real_distribution<> dist(-1., 1.);
+  const auto nn_generator = make_not_null(&generator);
```

Comment:
I'm not a huge fan of these extra variables for the not-nulls -- they slightly obscure the point of the `make_not_null` (to make unambiguous every function call that mutates the data), and they're only used once in this function.

Reply:
agreed, removed in the fixup